Automattic/jetpack
# RLT proxy for Likes and Notifications

Fixes #15184

#### Changes proposed in this Pull Request:

* Fix Like buttons in Safari by enabling cross-domain authentication.

#### Jetpack product discussion
pcNnmV-1L-p2

#### Does this pull request change what data or activity we track or use?
This provides the browser with a token, known as an RLT or "Remote Login Token", which is used to authorize Like and Notification requests, just like we currently do with cookies. The token is stored in localStorage for the Jetpack site, in the user's browser, and sent with some API requests. The reason for this change is that browsers are increasingly blocking "third-party" cookies - which includes any cookies normally set or sent by cross-origin iframes.

#### Testing instructions:

* In Chrome Incognito mode, or in Safari, attempt to interact with a Like or Comment Like button.
* You should be prompted to log into your WPCOM account in a popup.
* Upon successful login, the Like or Comment Like button should become interactive.
* Bonus: if you are logged into the site itself, and have toolbar notifications enabled, then that widget will now work as well - but only if you have Likes enabled.
